adapters/aws-sns.md

```diff
@@ -0,0 +1,25 @@
+# AWS Simple Notification Service CloudEvents Adapter
+
+This document describes how to convert
+[AWS SNS notifications](https://docs.aws.amazon.com/sns/latest/api/Welcome.html)
+into a CloudEvents.
+
+AWS SNS notification documentation:
+https://docs.aws.amazon.com/sns/latest/api/Welcome.html
+
+All SNS notifications are converted into CloudEvents using the
+same pattern as described in the following table:
+
+| CloudEvents Attribute | Value                                           |
+| :-------------------- | :---------------------------------------------- |
+| `id`                  | "x-amz-sns-message-id" value                    |
```

Comment:
We have talked at length that `id` is suppose to be a new uuid even if the upstream source provides an id.

Reply:
Is [this](https://github.com/cloudevents/spec/blob/master/spec.md#id) no longer accurate then?

> Producers MUST ensure that source + id is unique for each distinct event. If a duplicate event is re-sent (e.g. due to a network error) it MAY have the same id. Consumers MAY assume that Events with identical source and id are duplicates.

Upstream sources may also re-send events due to network errors. Doug's proposal allows those duplicates to be detectable downstream.

Reply:
We only need to create a UUID if the `x-amz-sns-message-id` is not unique per SNS event - I was assuming it was.  The docs say this:
```
A Universally Unique Identifier, unique for each message published. For a notification that
Amazon SNS resends during a retry, the message ID of the original message is used. 
```
So I think that should be ok.  @timbray would you agree?